modules/programs/gradle.nix

```diff
@@ -0,0 +1,134 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.gradle;
+  defaultHomeDirectory = ".gradle";
+  settingsFormat = pkgs.formats.javaProperties { };
+
+  initScript = types.submodule ({ config, ... }: {
+    options = {
+      name = mkOption {
+        type = types.str;
+        description = ''
+          The name of the init script. Has to end with either '.gradle' for
+          Groovy init scripts or '.init.gradle.kts' for Kotlin init scripts.
+          See https://docs.gradle.org/current/userguide/init_scripts.html for
+          more details.
+        '';
+      };
+      text = mkOption {
+        type = types.lines;
+        default =
+          if config.source != null then builtins.readFile config.source else "";
+        defaultText = literalExpression
+          "if source is defined, the content of source, otherwise empty";
+        description = ''
+          Text of the init script file.
+          If unset then the source option will be preferred.
+        '';
+      };
+
+      source = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path of the init script file to use.
+          If the text option is set, it will be preferred.
+        '';
+      };
+    };
```

Comment:
Add `config` section where source is populated with the generated file, see `file-type.nix`.

Reply:
… when `mkIf (config.text != null)`, that is… 🙂

Reply:
Wow, this was way more tricky than I anticipated. But it's done now.